src/lib.rs

```diff
@@ -495,6 +495,43 @@ impl Client {
         nonce: Option<String>,
         signer: &impl Signer,
         url: &str,
+    ) -> Result<BytesResponse, Error> {
+        let mut retries = 3;
+
+        let mut nonce = nonce;
+        loop {
+            let mut response = self.inner_post(payload, nonce.clone(), signer, url).await?;
+            if response.parts.status == StatusCode::BAD_REQUEST {
+                let body = response.body.into_bytes().await.map_err(Error::Other)?;
+                let problem: Problem = serde_json::from_slice(&body)?;
+                if problem
+                    .r#type
+                    .map(|t| t == "urn:ietf:params:acme:error:badNonce")
+                    .unwrap_or_default()
+                {
+                    retries -= 1;
+                    if retries != 0 {
+                        nonce = Some(self.nonce(None).await?);
```

Comment:
Acquiring a nonce here seems unnecessary -- we can just set it `None`, since `inner_post()` will take care of this for us?

Reply:
I realised that a new nonce is available in the `Replay-Nonce` header so have switched to grabbing that out instead. It fails gracefully, setting `nonce` to `None` if the header isn't there, but the header is required by the spec so should definitely be there. This way we avoid an HTTP request all together.